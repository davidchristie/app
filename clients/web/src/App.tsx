import { Box, CssBaseline } from "@material-ui/core";
import { ErrorBoundary } from "react-error-boundary";
import { BrowserRouter } from "react-router-dom";
import { ApiProvider } from "./api";
import { ErrorPage } from "./pages/ErrorPage";
import { Routes } from "./Routes";

export function App() {
  return (
    <ErrorBoundary FallbackComponent={ErrorPage}>
      <Box data-testid="App">
        <CssBaseline />
        <ApiProvider>
          <BrowserRouter>
            <Routes />
          </BrowserRouter>
        </ApiProvider>
      </Box>
    </ErrorBoundary>
  );
}
