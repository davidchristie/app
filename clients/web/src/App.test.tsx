import {
  render,
  screen,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import { App } from "./App";

describe("App", () => {
  describe("while loading", () => {
    it("renders correctly", () => {
      const { container } = render(<App />);
      expect(screen.getByTestId("App__loading")).toBeDefined();
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("once loading is complete", () => {
    it("renders correctly", async () => {
      const { container } = render(<App />);
      await waitForElementToBeRemoved(() =>
        screen.queryByTestId("App__loading")
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
