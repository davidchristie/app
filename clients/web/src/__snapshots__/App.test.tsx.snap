// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App once loading is complete renders correctly 1`] = `
<div
  data-testid="App"
>
  <header
    data-testid="Header"
  >
    <pre>
      {
  "user": {
    "id": "97406d59-7a49-4f1e-bb79-aba34cfcb405",
    "name": "Test User",
    "email": "test_user@email.com"
  }
}
    </pre>
  </header>
  <main
    style=""
  >
    <p>
      <a
        href="/api/v1/auth/github/authorize"
      >
        Sign in with GitHub
      </a>
    </p>
  </main>
</div>
`;

exports[`App while loading renders correctly 1`] = `
<div
  data-testid="App"
>
  <main
    style="display: none;"
  >
    <p>
      <a
        href="/api/v1/auth/github/authorize"
      >
        Sign in with GitHub
      </a>
    </p>
  </main>
  <div
    data-testid="App__loading"
  >
    Loading...
  </div>
</div>
`;
